app/services/foreman/multimessage_exception.rb

```diff
@@ -0,0 +1,30 @@
+module Foreman
+  class MultimessageException < ::Foreman::Exception
+    attr_accessor :messages
+
+    def initialize(messages_array, message, *params)
+      super(message, *params)
+      @messages = messages_array
+    end
+
+    def add_message(msg)
+      @messages << msg
+    end
+
+    def <<(msg)
+      add_message(msg)
+    end
+
+    def message
+      if @messages.size > 0
+        super + ": #{@messages.split(', ')}"
```

Comment:
I haven't followed precisely where this is used, but the colon should be part of the message preceding it for translation purposes.